CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivities.uc

```diff
@@ -701,83 +545,68 @@ static function int GetMissionDifficultyFromMonthPlusTemplate (XComGameState_Act
 	return Difficulty;
 }
 
-static function PreMissionSetup_DE (XComGameState NewGameState, XComGameState_Activity ActivityState)
+static function WaitSetup (XComGameState NewGameState, XComGameState_Activity ActivityState)
 {
-	local XComGameState_MissionSite MissionState;
-	local XComGameState_DarkEvent DarkEventState;
+	local int SecondsWaitDuration;
+	local XComGameState_Activity_Wait WaitActivity;
 
-	MissionState = class'X2Helper_Infiltration'.static.GetMissionStateFromActivity(ActivityState);
-	DarkEventState = ActivityState.GetActivityChain().GetChainDarkEvent();
+	`CI_Log("Setting up wait stage");
 
-	if (DarkEventState != none)
+	class'X2Helper_Infiltration'.static.GetWaitPeriodDuration(
+		class'XComGameState_ActivityChainSpawner'.default.MinGenericWaitDays, 
+		class'XComGameState_ActivityChainSpawner'.default.MaxGenericWaitDays, 
+		SecondsWaitDuration);
+	
+	
+	WaitActivity = XComGameState_Activity_Wait(ActivityState);
+	// No need to call NewGameState.ModifyStateObject here as SetupStage is passed an already modified state
+
+	if (WaitActivity == none)
 	{
-		MissionState.DarkEvent = DarkEventState.GetReference();
+		`RedScreen(ActivityState.GetMyTemplateName() $ " is not a wait activity but calls WaitSetup!");
+		return;
 	}
+
+	WaitActivity.ProgressAt = `STRATEGYRULES.GameTime;
+	class'X2StrategyGameRulesetDataStructures'.static.AddTime(WaitActivity.ProgressAt, SecondsWaitDuration);
 }
 
-static function string GetUnitDetails (XComGameState_Activity ActivityState, XComGameState_Reward RewardState)
+static function DarkEventWaitSetup (XComGameState NewGameState, XComGameState_Activity ActivityState)
```

Comment:
The previous logic allocated "windows" to each DE, to prevent the chains from showing up at once. Unless I'm missing something, this implementation allows for all 3 (example) chains to show up at the same time, depending on how the dice rolls

Reply:
Whoops, I did not realize that was what the old logic did.